### DISCUSSION TITLE
how do i make nestedArray to render like default values when i click append button

### DISCUSSION THREAD

**Discussion by Raju-2101** [react-hook-form/react-hook-form#2325] [Q&A]
https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-x7btr

in the above example how do i make nestedArray with useFieldArray1 to render like default values when i click append button in fieldArray.js. i tried append(defaultValues) but it doesn't work

**Comment by bluebill1049** [comment]
you can use `useEffect` on the nested FieldArray?

**Reply by Raju-2101** [reply]
how do i do that

**Reply by bluebill1049** [reply]
```
useEffect(() => {
  append({ yourDefaultValue })
}, [append])
```

**Reply by Raju-2101** [reply]
thx
